mycore-base/src/main/java/org/mycore/services/queuedjob/MCRJobMaster.java

```diff
@@ -178,74 +177,76 @@ protected void beforeExecute(Thread t, Runnable r) {
-                                try {
-                                    transaction.rollback();
-                                } catch (RuntimeException re) {
-                                    LOGGER.warn("Could not rollback transaction.", re);
+                                transaction.commit();
+                            } catch (RollbackException e) {
+                                LOGGER.error("Error while getting next job.", e);
+                                if (transaction != null) {
+                                    try {
+                                        transaction.rollback();
+                                    } catch (RuntimeException re) {
+                                        LOGGER.warn("Could not rollback transaction.", re);
+                                    }
                                 }
+                            } finally {
+                                em.close();
                             }
-                        } finally {
-                            em.close();
-                        }
-                        if (job != null && action != null && action.isActivated() && !jobServe.isShutdown()) {
-                            LOGGER.info("Creating:" + job);
-                            jobServe.execute(new MCRJobThread(job));
-                        } else {
-                            try {
-                                synchronized (JOB_QUEUE) {
-                                    if (running) {
-                                        LOGGER.debug("No job in queue going to sleep");
-                                        //fixes a race conditioned deadlock situation
-                                        //do not wait longer than 60 sec. for a new MCRJob
-                                        JOB_QUEUE.wait(60000);
+                            if (job != null && action != null && action.isActivated() && !jobServe.isShutdown()) {
+                                LOGGER.info("Creating:" + job);
+                                jobServe.execute(new MCRJobThread(job));
+                            } else {
+                                try {
+                                    synchronized (JOB_QUEUE) {
+                                        if (running) {
+                                            LOGGER.debug("No job in queue going to sleep");
+                                            //fixes a race conditioned deadlock situation
+                                            //do not wait longer than 60 sec. for a new MCRJob
+                                            JOB_QUEUE.wait(60000);
+                                        }
                                     }
+                                } catch (InterruptedException e) {
+                                    LOGGER.error("Job thread was interrupted.", e);
                                 }
-                            } catch (InterruptedException e) {
```

Comment:
Database Exception may result in to much logs to handle, rest here for a while, please ;-)